add C implementation for `@stdlib/math/base/special/frexp` (#650)
stdlib-js/stdlib: lib/node_modules/@stdlib/math/base/special/frexp/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/math/base/special/frexp.h"
```

#### stdlib_base_frexp( x, frac, exp )

Splits a [double-precision floating-point number][ieee754] into a normalized fraction and an integer power of two.

```c
#include <stdint.h>

stdlib_base_frexp( 4.0, &frac, &exp );
```

The function accepts the following arguments:

-   **x**: `[in] float` input value.

> -   **frac**: `[out] float*` destination to a normalized fraction.
> -   **exp**: `[out] int32_t*` destination to an integer power of two.

Reviewer: ```suggestion
-   **frac**: `[out] float*` destination for the normalized fraction.
-   **exp**: `[out] int32_t*` destination for the integer power of two.
```
